src/assets/Keycloak/oCIS-with-Keycloak.md

```diff
@@ -0,0 +1,195 @@
+    depends_on:
+      - postgres
+
+volumes:
+  postgres_data:
+```
+
+Save the above code in `compose.yaml` file locally and run docker command `docker compose up` in a terminal.
+
+Docker compose serves three services:
+- oCIS on `https://host.docker.internal:9200`
+- Keycloak on `https://host.docker.internal:8443`
+- Traefik dashboard on `http://host.docker.internal:8080/`
+
+### Manual Steps For Setting Up oCIS With Keycloak
+1. Go to oCIS url: `https://host.docker.internal:9200/`
+
+![oCIS](/src/assets/Keycloak/images/oCIS-Keycloak/oCIS-Keycloak-UI-setup-1.png)
+
+2. Go to Keycloak url: `https://host.docker.internal:8443/`
+
+![keycloak](/src/assets/Keycloak/images/oCIS-Keycloak/oCIS-Keycloak-UI-setup-2.png)
+
+3. Create new realm: `oCIS`
+
+![create realm](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-1.png)
+![create realm with name `oCIS`](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-2.png)
+> Note: the realm name should be the name provided in the oCIS environment variable: `OCIS_OIDC_ISSUER: https://host.docker.internal:8443/realms/oCIS`
+
+4. Add a new client:
+
+![client registration](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-1.png)
+
+- Client ID: `web`
+
+![add client](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-2.png)
+
+- Root URL: `https://host.docker.internal:9200`
+
+![add root url](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-3.png)
+
+5. Add realm roles:
+
+- Role name: `ocisAdmin`
+
+![add realm role](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-role.png)
+
+> Note: oCIS has defined roles name that has been documented under [Automatic Quota Assignments](https://owncloud.dev/services/proxy/#automatic-role-assignments). So, we can't create the role with a random name.
+
+6. Create a new user:
+
+- With `username: admin` and other information:
+
+![create user](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-1.png)
+![add user information](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-2.png)
+
+- Create password
+
+![create password](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-3.png)
+![setup password for user](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-4.png)
+
+- Role Mapping: Assign `ocisAdmin` role
+
+![assign role](/src/assets/Keycloak/images/oCIS-Keycloak/assign-role-to-user-1.png)
+
+![assign role to user](/src/assets/Keycloak/images/oCIS-Keycloak/asign-role-to-user-2.png)
+
+7. Update `roles` Client scope: `Client Scopes` -> `roles`
+
+- `Mappers` -> `realm roles` -> `Token Claim Name=roles`
+![update-roles](/src/assets/Keycloak/images/oCIS-Keycloak/update-roles.png)
+> Note: oCIS default role_claim is 'roles'
+
+8. Go to url `https://host.docker.internal:9200/`, reload if already opened
+![login to oCIS with keycloak](/src/assets/Keycloak/images/oCIS-Keycloak/oCIS-Keycloak-login.png)
+Enter `username: admin` and `password: admin` to log in oCIS
```

Comment:
I still think that there is a "to" missing in this sentence (log in to sth), see https://github.com/JankariTech/blog/pull/122#discussion_r1601337686. Did you check this with @phil-davis or with the English teacher?

Reply:
Maybe these links are useful to understand when a preposition is needed before an object in combination with login:
https://promova.com/confusing-words/log-in-login
https://promova.com/confusing-words/log-in-log-on

I'm sure the English teacher can explain it in more detail to you how this applies to your text.